### ISSUE TITLE
H5MD reader: skip the first frame

### ISSUE THREAD

**Issue by junghans** [votca/csg#535]
If I run:
```
csg_stat --nt 1 --options settings.xml --top topol.xml --trj traj.h5 --begin 0 --first-frame 0
```
with these [inputs](https://github.com/votca/csg/files/4664174/input.zip), the output (`CG-CG.dist.new`) is all zeros, even though the positions in `traj.h5` looks ok.

Longer trajectories works fine as well.

**Comment by junghans** [comment]
In contrast:
```
csg_stat --nt 1 --options settings.xml --top topol.xml --trj spce.gro --begin 0 --first-frame 0
```
works.

**Comment by jkrajniak** [comment]
I have debug a bit the example and the simulation script (so nice it is included in h5md file). I think the zeros are not because of not reading the first frame but because the .h5 uses angstroms and the settings.xml are in nm? If you change `<max>` to 9.0 then it calculates distribution.

**Comment by junghans** [comment]
Good catch it seems we forgot to fix the reader after merging https://github.com/votca/csg-tutorials/pull/34. The simulation is now done in Angstrom, but the reader still thinks numbers are in nm.

Is there a field to store the units in the h5md format?

**Comment by jkrajniak** [comment]
There is a module called `units` which holds the information about the unit system (https://nongnu.org/h5md/modules/units.html) This is nothing more than an `unit` attribute assigned to the dataset. However, if I am correct Espresso does not use this attribute to store the units (which I admit, could be beneficial)

**Comment by junghans** [comment]
VOTCA internal units are nm, hence the missing factor 10.

As a workaround we could assume h5md is in Angstrom if no unit is given.

@JoshuaSBrown what do you think?

@jkrajniak is there are way to set the unit manually in the espresso script?

**Comment by jkrajniak** [comment]
> As a workaround we could assume h5md is in Angstrom if no unit is given.

If VOTCA expects that length is expressed in nm then I would assume that the numbers in
H5MD are in nm units (so no conversion) and do the conversion only if the `units` is enabled. Just to follow the principle of garbage-in-garbage-out.
> 
> @JoshuaSBrown what do you think?
> 
> @jkrajniak is there are way to set the unit manually in the espresso script?

AFAIK not without modifying the `h5md.H5md` module in Espresso

**Comment by JoshuaSBrown** [comment]
I think if there is a units attribute in h5md it should be used, and if it is not specified either a warning should be thrown indicating that a default is being used or an error should be thrown.

**Comment by JoshuaSBrown** [comment]
Since

**Comment by junghans** [comment]
@jngrad @fweik @RudolfWeeber could you expose the unit property for the h5md reader to the user in espresso?

**Comment by jngrad** [comment]
@junghans espresso simulation data is typically dimensionless. I just tried to create a new field `"particles/atoms/id/time/unit", 1, type_cstring}` with `h5xx::datatype type_cstring = h5xx::datatype(H5T_NATIVE_CHAR)`, but I get an exception `creating dataset "particles/atoms/id/time/unit"` from [5xx/dataset/dataset.hpp#L145](https://github.com/h5md/h5xx/blob/24c87aac1c0f0d992c98e4a655ad63131183b4f5/h5xx/dataset/dataset.hpp#L145) without information about what's wrong.

**Comment by junghans** [comment]
> @junghans espresso simulation data is typically dimensionless. I just tried to create a new field `"particles/atoms/id/time/unit", 1, type_cstring}` with `h5xx::datatype type_cstring = h5xx::datatype(H5T_NATIVE_CHAR)`, but I get an exception `creating dataset "particles/atoms/id/time/unit"` from [5xx/dataset/dataset.hpp#L145](https://github.com/h5md/h5xx/blob/24c87aac1c0f0d992c98e4a655ad63131183b4f5/h5xx/dataset/dataset.hpp#L145) without information about what's wrong.

That is a question foro @jkrajniak ;-)

**Comment by junghans** [comment]
ping @jkrajniak can you have a look into the error @jngrad found in `h5xx` in espresso?

**Comment by jkrajniak** [comment]
I've just added PR to espresso with the ability to store units alongside with the H5MD file (https://github.com/espressomd/espresso/pull/3751)

**Comment by junghans** [comment]
Writing of units was added to espresso in espressomd/espresso#3751, reading is added in #549, now we just need to wait for the next espresso release and update our tutorial.

**Comment by junghans** [comment]
Espresso-4.1.3 is out, but it doesn't include the h5md unit fix, waiting for the next release.

In the tutorials there is only one line that needs to be added, something like:
https://github.com/espressomd/espresso/blob/9b62aa2bfc1eb808ad71f4640f65af93e99a2bba/samples/h5md.py#L46
